contracts/dev/functions/FunctionsClient.sol

```diff
@@ -10,8 +10,8 @@ import "../interfaces/FunctionsOracleInterface.sol";
  * @notice Contract writers can inherit this contract in order to create Chainlink Functions requests
  */
 abstract contract FunctionsClient is FunctionsClientInterface {
-  FunctionsOracleInterface private s_oracle;
-  mapping(bytes32 => address) private s_pendingRequests;
+  FunctionsOracleInterface internal s_oracle;
+  mapping(bytes32 => address) internal s_pendingRequests;
```

Comment:
These should be `internal`, not `private` as it allows a `FunctionsConsumer` contract to use these values directly.  This is necessary for `AutomatedFunctionsConsumer` to avoid running the expensive `encodeCBOR` for every request triggered by Automation.

Reply:
See the `AutomatedFunctionsConsumer` contract or `deployAutoClient.js` for details.

Reply:
Please remember to port contract changes to core repo afterwards (or even better before merging this PR).

Reply:
Let's do core repo first, opened here: https://github.com/smartcontractkit/chainlink/pull/8429